### ISSUE TITLE
ErrDocumentIfNotAttached occurs when trying to remove documents in Dashboard

### ISSUE THREAD

**Issue by hackerwins** [yorkie-team/yorkie#758]
**What happened**:

`ErrDocumentIfNotAttached` occurs when trying to remove documents in Dashboard.

![스크린샷 2024-01-11 오전 11 00 03](https://github.com/yorkie-team/yorkie/assets/2059311/12932915-4d3d-4337-8681-d0fcb7cf7df2)

**What you expected to happen**:

A document should be removed if it is not attached to clients.

**How to reproduce it (as minimally and precisely as possible)**:

Try to remove any documents in Dashboard.

**Anything else we need to know?**:

It is good to check the reason for the problem is `documentIfNotAttached` flag: https://github.com/yorkie-team/yorkie/pull/560

**Environment**:
- Operating system: MacOS
- Browser and version: Chrome latest
- Yorkie version (use `yorkie version`): v0.4.12
- Yorkie JS SDK version: v0.4.12

**Comment by fourjae** [comment]
Good, may i try this issue?

**Comment by raararaara** [comment]
This issue occurred because the document was not properly detached when the client was deactivated. This has been modified in #907 to ensure that document detachment is correctly reflected when deactivating a client.

After these modifications are applied, correction logic is required for clientInfos with deactivated status that already have attached documents.

Through the query below, you can search for documents that meet the conditions and perform modifications on each.
```ts
db.clients.find({
  status: 'deactivated',
  documents: { $ne: null },
  $expr: {
    $gt: [{
      $size: {
        $filter: {
          input: { $objectToArray: "$documents" },
          as: "doc",
          cond: { $eq: ["$$doc.v.status", "attached"] }
        }
      }
    }, 0]
  }
}).forEach(doc => {
  var updatedDocuments = {};

  for (var key in doc.documents) {
    if (doc.documents[key].status === 'attached') {
      updatedDocuments[key] = {
        status: 'detached',
        client_seq: 0,
        server_seq: 0
      };
    } else {
      updatedDocuments[key] = doc.documents[key];
    }
  }
  
  var currentTime = new Date();

  db.clients.updateOne(
    { _id: doc._id },
    { 
      $set: { 
        documents: updatedDocuments,
        updated_at: currentTime
      } 
    }
  );
});

```

**Comment by hackerwins** [comment]
This was fixed by #907
